pkg/curl/schemes.go

```diff
@@ -147,6 +155,18 @@ func (s Schemes) Fetch(ctx context.Context, u *url.URL) (File, error) {
 	return &file{ReaderAt: r, url: u}, nil
 }
 
+func (s Schemes) FetchWithoutCache(ctx context.Context, u *url.URL) (File, error) {
+	fg, ok := s[u.Scheme]
+	if !ok {
+		return nil, &URLError{URL: u, Err: ErrNoSuchScheme}
+	}
+	r, err := fg.FetchWithoutCache(ctx, u)
+	if err != nil {
+		return nil, &URLError{URL: u, Err: err}
+	}
+	return &file{Reader: r, url: u}, nil
```

Comment:
This doesn't seem right: https://play.golang.org/p/nwtVj_moQvu

It just panics if you try to call `ReaderAt`. How does the user know to use `Reader` or `ReaderAt`?

Reply:
read the code ? `&file{Reader: r, url: u},`

Arguably, File need to implement both ReaderAt and Reader.

Reply:
Yes, I think this should be implemented in such a way that all the functions in the interface are implemented. Either remove `Reader` or `ReaderAt`, or implement both.

As a caller of the function, it is quite expected to receive an interface where some of the functions are not implemented and panic.

Reply:
Done.

but in a slight different way, and on the same line. The File interface is updated.

I tried, via wrappers, complete `io.Reader` for cached `io.ReaderAt`, and same for non-cache `io.Reader` with `io.Reader`, but then took a step back and think why we need to do that. It was all due to `File` interface says so. Looking at what `File` was defined for, it was simply meant to provide a way to string print original URL. So instead of adding additional code to force implement both `io.Reader` and `io.ReaderAt` for both, we can simply remove the string conformance requirement from `File`, but keeping `URL()` and adding `fmt.Stringer`. 

letting implementers be different at either returning `io.ReaderAt` or `io.Reader`, which will be used, than implementing the other half for each, which will not be used practically.

If we are going to need it in future, one can simply move either `io.ReaderAt` or `io.Reader` to common interface, and ask both to implement it.